frontend/src/metabase/lib/formatting/date.tsx

```diff
@@ -634,6 +630,29 @@ export function formatDateTimeRangeWithUnit(
     return String(start);
   }
 
+  const formatDate = (date: Moment, formatStr: string) => {
+    const yearRegex = /,*\s*YYYY/;
+    const dayRegex = /[\s\S]+,*\s*YYYY,*\s*/;
+
+    // day display is hide-able (if date being compared to is same day)
+    if (options.removeDay) {
```

Comment:
The new `removeDay` and  `removeYear` options assume `formatStr` date format and probably won't work on the other date formats. 

I would assume that instead of replacing formatStr we need to compose it right away without a day or a year before it gets to this function

Reply:
It would be also nice to add more unit test son `date.tsx` specifically to verify more combinations on different formats since it is our common formatting code and it should have good coverage